### ISSUE TITLE
Offer filter to show intermediate items

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#4944]
# Summary
Offer a way to show intermediate items leading to filtered items

# Motivation

Currently I can filter for "pods" and it'll hide the intermediate deployment and replicaset along with the inactive replicasets.
It'd be nice to be able to see the active deployment and replicaset leading to the "pods" but not the ones that don't lead to pods.

https://cd.apps.argoproj.io/applications/jenkins?resource=kind%3APod

Possible today (too much noise):
![unfiltered](https://user-images.githubusercontent.com/2119212/100778921-9ec9ad80-33d5-11eb-9369-06c5c2b85925.png)
Possible today (omits too much):
![current filter for pods](https://user-images.githubusercontent.com/2119212/100779266-244d5d80-33d6-11eb-84b3-069c8ddc645c.png)

The circled items would be nice to see (The x'd items would be nice to not see):
![changes](https://user-images.githubusercontent.com/2119212/100779230-11d32400-33d6-11eb-9f42-eb61a134e8e0.png)

# Proposal

An extra check box in filters:
* [x] show intermediate nodes

**Comment by rbreeze** [comment]
In an upcoming feature called pod view, we'll be adding something similar. You'll be able to sort pods by their parent resources, or top level resource. https://github.com/argoproj/argo-cd/pull/4497

Will this solve your use case? Or do you need to be able to see every node in the path from app to pod?

**Comment by jessesuen** [comment]
I agree with @rbreeze. the pod view is intended to solve this use case.

**Comment by jsoref** [comment]
I want to be able to see the live manifest for the deployment/replicaset

**Comment by jsoref** [comment]
To be clear, I'm using `node` to mean `graph element along chain` not `kubernetes node`. And offhand, yes, I'm pretty sure I want to see all of the parent containers to a resource for this use case.

**Comment by jsoref** [comment]
Here's a fancier example:
https://cd.apps.argoproj.io/applications/argo-events?view=tree
![image](https://user-images.githubusercontent.com/2119212/107430599-5cb69980-6af3-11eb-9d83-6051d9516c21.png)
![image](https://user-images.githubusercontent.com/2119212/107430846-b3bc6e80-6af3-11eb-9750-8d5ff67bc911.png)

https://cd.apps.argoproj.io/applications/argo-events?view=pods
By top level resource:
![image](https://user-images.githubusercontent.com/2119212/107430970-d484c400-6af3-11eb-9c3c-9a445968e129.png)
This gives me access to `Calendar`:
https://cd.apps.argoproj.io/applications/argo-events?view=pods&node=argoproj.io%2FEventSource%2Fargo%2Fcalendar%2F0
![image](https://user-images.githubusercontent.com/2119212/107431044-ebc3b180-6af3-11eb-8c90-6e7056b04b6b.png)
By parent resource:
![image](https://user-images.githubusercontent.com/2119212/107431091-fb42fa80-6af3-11eb-9c3d-560f05788272.png)
This gives me access to `RS`:
![image](https://user-images.githubusercontent.com/2119212/107431130-072ebc80-6af4-11eb-8ea4-77fbd80a80e3.png)

This gives me access to the node (once I tried all 8 other pods -- because I had 9 guesses and guessed wrong all possible times before finding the right node!)
![image](https://user-images.githubusercontent.com/2119212/107431212-1f9ed700-6af4-11eb-8283-9865860ac88d.png)
Also, there's no metadata for the node, which is surprising.

What I can't access at all here is the deployment.

**Comment by rbreeze** [comment]
> Also, there's no metadata for the node, which is surprising.
 
What do you mean by metadata? 

In this specific example, I see what you mean - because of the long chain between the EventSource and Pod, you can't see one of those resources in that chain in pod view. Why don't you use the tree view to access the deployment?

Most of the time deployments *do* show up in pod view top level resource. This is somewhat of an edge case.

**Comment by jsoref** [comment]
By metadata, I mean that I can't click it to get a right covering sheet (pictured in roughly every second picture in that comment -- I'm not sure what they're actually called), unlike almost all other objects.

The tree view generally has lots and lots of replicasets with no pods, which makes it painful to view, hence my original request. And sometimes, there will be a handful of RS each with pods.

While this is an edge case, I kinda expected it to be present, and the sample dataset provided it, I didn't even have to look hard in the dataset! (It took me more effort to find the right pod in the nodes view!)

But again, even when there isn't an intermediate, there's the general problem of not being able to easily associate the pod to both its parent and grandparent.

The tree I pictured in the comment is a good example of how painful the tree fanout is -- see the bottom w/ the multiple empty `RS`s.
